marketplaceItems/drawApp/draw_app.js

```diff
@@ -0,0 +1,504 @@
+                if (!polyLine) {
+                    polyLine = Entities.addEntity({
+                        type: "PolyLine",
+                        name: "Draw App Polyline",
+                        position: lineStartPosition,
+                        linePoints: linePoints,
+                        normals: lineNormals,
+                        strokeWidths: lineStrokeWidths,
+                        color: { red: 255, green: 255, blue: 255 },
+                        isUVModeStretch: true,
+                        lifetime: DECAY_TIME_S,
+                        collisionless: true
+                    }, 'avatar');
+                } else {
+                    var lineProperties = Entities.getEntityProperties(polyLine, ['linePoints', 'normals', 
+                        'strokeWidths', 'age']);
+                    var linePointsCount = lineProperties.linePoints.length;
+                    if (linePointsCount > MAX_LINE_POINTS) {
+                        var lastPointDisplacement = lineProperties.linePoints[linePointsCount - 1];
+                        linePoints = [ lastPointDisplacement, displacementFromStart ];
+                        lineNormals = [ DEFAULT_NORMAL, DEFAULT_NORMAL ];
+                        lineStrokeWidths = [ DEFAULT_STROKE_WIDTH_M, DEFAULT_STROKE_WIDTH_M ];
+                        polyLine = Entities.addEntity({
+                            type: "PolyLine",
+                            name: "Draw App Polyline",
+                            position: previousLinePoint,
+                            linePoints: linePoints,
+                            normals: lineNormals,
+                            strokeWidths: lineStrokeWidths,
+                            color: { red: 255, green: 255, blue: 255 },
+                            isUVModeStretch: true,
+                            lifetime: DECAY_TIME_S,
+                            collisionless: true
+                        }, 'avatar');
+                    } else {
+                        lineProperties.linePoints.push(displacementFromStart);
+                        lineProperties.normals.push(DEFAULT_NORMAL);
+                        lineProperties.strokeWidths.push(DEFAULT_STROKE_WIDTH_M);
+                        Entities.editEntity(polyLine, {
+                            linePoints: lineProperties.linePoints,
+                            normals: lineProperties.normals,
+                            strokeWidths: lineProperties.strokeWidths,
+                            lifetime: lineProperties.age + DECAY_TIME_S
+                        });
+                    }
+                }
+            }
+        }
+    }
+
+    /* ON MOUSE RELEASE: Stop checking distance cursor has moved and update current polyline lifetime to decay in 1 
+    minute. */
+    function mouseReleased(event) {
+        if (event.button === "LEFT") {
+            polyLine = null;
+            linePoints = [ {x: 0, y: 0, z: 0 } ];
```

Comment:
please remove spaces after `[` and before `]` in this line and the two lines below